java/monkey/src/main/java/Lexer.java

```diff
@@ -40,18 +42,10 @@ public Token nextToken() {
             case '>' -> TokenType.GT.token();
             case '<' -> TokenType.LT.token();
             case '\0' -> TokenType.EOF.token();
-            default -> TokenType.ILLEGAL.createToken(String.valueOf(this.getCc()));
-        };
 
-        if (token.type() != TokenType.ILLEGAL) {
```

Comment:
Could we accomplish this without the `pos -1`?

Reply:
We could write
```java
private String indent() {
        int pos = this.pos;
        while (this.isLetter(this.getCc())) {
            this.advance();
        }
        return this.input.substring(pos - 1, this.pos);
    }
```

Or go back to what we were doing before (selectively advancing the pointer) but my preference for this new way in the comment above.